src/Symfony/Component/Workflow/Dumper/PlantUmlDumper.php

```diff
@@ -0,0 +1,198 @@
+        $code = $this->initialize($options);
+        foreach ($definition->getPlaces() as $place) {
+            $code[] =
+                "state {$place}".
+                ($definition->getInitialPlace() === $place ? ' <<'.self::INITIAL.'>>' : '').
+                ($marking && $marking->has($place) ? ' <<'.self::MARKED.'>>' : '');
+        }
+        if ($this->isSquareTransitionType()) {
+            foreach ($definition->getTransitions() as $transition) {
+                $code[] = "agent {$transition->getName()}";
+            }
+        }
+        foreach ($definition->getTransitions() as $transition) {
+            foreach ($transition->getFroms() as $from) {
+                foreach ($transition->getTos() as $to) {
+                    if ($this->isSquareTransitionType()) {
+                        $lines = array(
+                            "{$from} --> {$transition->getName()}",
+                            "{$transition->getName()} --> {$to}",
+                        );
+                        foreach ($lines as $line) {
+                            if (!in_array($line, $code)) {
+                                $code[] = $line;
+                            }
+                        }
+                    } else {
+                        $code[] = "{$from} --> {$to}: {$transition->getName()}";
+                    }
+                }
+            }
+        }
+
+        return $this->startPuml($options).$this->getLines($code).$this->endPuml($options);
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return string
+     */
+    protected function startPuml(array $options)
+    {
+        $start = '@startuml'.PHP_EOL;
+
+        if ($this->isSquareTransitionType()) {
+            $start .= 'allow_mixing'.PHP_EOL;
+        }
+
+        if (isset($options['nofooter']) && true === $options['nofooter']) {
+            return $start;
+        }
+
+        return $start.static::SYMFONY_LOGO.PHP_EOL;
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return string
+     */
+    protected function endPuml(array $options)
+    {
+        $end = PHP_EOL.'@enduml';
+        if (isset($options['nofooter']) && true === $options['nofooter']) {
+            return $end;
+        }
+
+        return PHP_EOL.'footer \nGenerated by <$sf_logo> **Workflow Component** and **PlantUML**'.$end;
+    }
+
+    /**
+     * @param array $code
+     *
+     * @return string
+     */
+    protected function getLines(array $code)
+    {
+        return implode(PHP_EOL, $code);
+    }
+
+    /**
+     * @param array $options
+     *
+     * @return array
+     */
+    protected function initialize(array $options)
+    {
+        $code = array();
+        if (isset($options['title'])) {
+            $code[] = "title {$options['title']}";
+        }
+        if (isset($options['name'])) {
+            $code[] = "title {$options['name']}";
+        }
+        if (isset($options['skinparams']) && count($options['skinparams']) > 0) {
```

Comment:
I don't think you need to count here

Reply:
correct I replace with is_array to ensure the foreach won't fail if someone send something else